files/image_config/rsyslog/rsyslog-config.sh

```diff
@@ -17,7 +17,10 @@ if [[ ($NUM_ASIC -gt 1) ]]; then
 else
     udp_server_ip=$(ip -j -4 addr list lo scope host | jq -r -M '.[0].addr_info[0].local')
 fi
+hostname=$(hostname)
```

Comment:
>hostname [](http://example.com/codeflow?start=0&length=8)

There is DEVICE_METADATA/hostname in ConfigDB. Could you check which one is better?

Reply:
it is better to take from linux than DEVICE_METADATA/hostname. 
example: DB can be empty if not configured with CLI. hostname can also be dynamically assigned using DHCP.

Reply:
If DEVICE_METADATA/hostname has value, it should be preferred.

Reply:
The hostname can be dynamically assigned by DHCP, in that case, DB value may contain the wrong hostname.
Plus `DEVICE_METADATA` is a configuration and not a STATE, once data is placed to DEVICE_METADATA it will update the system hostname according to https://github.com/sonic-net/sonic-host-services/blob/bc08806b64002c506b8401eae5d9e1c760651e49/scripts/hostcfgd#L1568
So in my opinion it is better to take it from the `hostname` utility as the most appropriate one.